dependencies.yaml

```
@@ -0,0 +1,127 @@
+# Dependency list for https://github.com/rapidsai/dependency-file-generator
+files:
+  all:
+    output: none
+    includes:
+      - build_python
+      - checks
+      - cudatoolkit
+      - py_version
+      - run
+      - test_cpp
+      - test_python
+  test_cpp:
+    output: none
+    includes:
+      - cudatoolkit
+      - test_cpp
+  test_python:
+    output: none
+    includes:
+      - cudatoolkit
+      - py_version
+      - test_python
+  checks:
+    output: none
+    includes:
+      - checks
+      - py_version
+channels:
+  - rapidsai
+  - rapidsai-nightly
+  - dask/label/dev
+  - conda-forge
+  - nvidia
+dependencies:
+  build_python:
+    common:
+      - output_types: [conda, requirements, pyproject]
+        packages:
+          - &cmake_ver cmake>=3.23.1,!=3.25.0
+          - librmm==23.4.*
+      - output_types: conda
+        packages:
+          - fmt>=9.1.0,<10
+          - &gtest gtest==1.10.0.*
+          - &gmock gmock==1.10.0.*
+          - spdlog>=1.11.0,<1.12
+          - cython>=0.29,<0.30
+          - numpy>=1.21
+          - rmm==23.4.*
+  checks:
+    common:
+      - output_types: [conda, requirements]
+        packages:
+          - pre-commit
+  cudatoolkit:
+    specific:
+      - output_types: conda
+        matrices:
+          - matrix:
+              cuda: "11.2"
+            packages:
+              - cudatoolkit=11.2
+          - matrix:
+              cuda: "11.4"
+            packages:
+              - cudatoolkit=11.4
+          - matrix:
+              cuda: "11.5"
+            packages:
+              - cudatoolkit=11.5
+          - matrix:
+              cuda: "11.8"
+            packages:
+              - cudatoolkit=11.8
+  py_version:
+    specific:
+      - output_types: conda
+        matrices:
+          - matrix:
+              py: "3.8"
+            packages:
+              - python=3.8
+          - matrix:
+              py: "3.9"
+            packages:
+              - python=3.9
+          - matrix:
+              py: "3.10"
+            packages:
+              - python=3.10
+          - matrix:
+            packages:
+              - python>=3.8,<3.11
+  run:
+    common:
+      - output_types: [conda, requirements]
```

Some additional work is probably needed to make this file compatible with generating the build/run/test requirements in `pyproject.toml`. Happy to defer that to a later PR if we can file an issue.
```suggestion
      - output_types: [conda, requirements, pyproject]
```